### ISSUE TITLE
Exact syncing of separate audio

### ISSUE THREAD

**Issue by akaihola** [jliljebl/flowblade-forum#56]
I'm shooting and recording a music performance. For video, I'm using a phone. For audio, I'm using a high-quality portable recorder which is not connected to the phone. I'm importing both the video and the separate audio file from the recorder into Flowblade.

To get the video and audio in sync, I'm first manually aligning them by observing the audio amplitude preview visible in both the video and the audio track. I then do `Select Clip to Audio Sync With...` in the video track context menu, and then select the audio track. Flowblade suggests to move the tracks a few frames, and I accept this with `Do Audio Sync Move Edit`. Finally, I mute audio on the video track.

Unfortunately the video and audio aren't still synced. And if I try to move the audio track manually, it moves in one video frame steps instead of e.g. millisecond accuracy.

Is there any way to move or trim the audio track with higher accuracy to get perfect video/audio synchronization?

**Comment by jliljebl** [comment]
>Is there any way to move or trim the audio track with higher accuracy to get perfect video/audio synchronization?

No, 1 frame accuracy is best Flowblade has. I don't think MLT or other libraries used to build Flowblade offer any way to do this, so there is no path to adding this currently.

Having done some commercial rock videos in the past I remember that sync was often clearly between frames for some shots, but when everything was edited together getting to the closest frame was good enough for the end product to look fine.

**Comment by akaihola** [comment]
Thanks @jliljebl. One frame accuracy definitely isn't enough for piano or drums. ~~I found that KDEnlive can do this, so for the time being I'm using it instead.~~

**Comment by jliljebl** [comment]
> I found that KDEnlive can do this, 

@akaihola This is actually useful information. This kind of feature can probably be quite easily copied, I'll give it look sometime.

There is no need to close these, some other user might be interested in this subject too.

**Comment by akaihola** [comment]
It seems I was just lucky with KDEnlive – my audio just happened to align nicely with the frames.

Here's a work-around I found to get perfect sync:

- make sure to start audio-only recording slightly before video recording
- do a hand clap right after both devices have started recording
- import both the audio-only track as well as the video `.mp4` into Audacity (it will use only the audio and ignore the video)
- move the video track to the right using the Time Shift Tool so the tracks are approximately aligned by comparing them visually
- zoom in all the way at the hand clap, and align tracks perfectly
- use the Selection Tool to mark the extra part at the beginning of the audio-only track by selecting from the beginning of the video track all the way left and hitting `Delete` (snapping helps to mark exactly – start the selection when you see a yellow vertical line)
- remove the video track
- export the audio track to a new `.wav` file
- import the `.wav` file to your video editor
- now edits to both the video and the separate audio clips line up perfectly to frames

**Comment by mkemper** [comment]
Hi, I am a newbie with flowblade and I am trying to create a music video. My experience is, that it is a bit difficult to get video and audio synced. It would be helpful to have a more detailed view on the wave curve I guess. By this one could more easily detect pause etc.

**Comment by jliljebl** [comment]
I'll make a note that this area could be improved. The audio levels display is not very detailed, that's true and more detailed displayed data would be useful in some scenarios, so we will try to get improvement on this.
